### ISSUE TITLE
consul_acl problem (healthchecks, proxy acl, intentions, etc...)

### ISSUE THREAD

**Issue by zhenik** [Skatteetaten/terraform-nomad-minio#26]
In context of #344 https://github.com/fredrikhgrelland/vagrant-hashistack/issues/344

I made some research for specific settings, when
```
consul_acl=true
consul_acl_default_policy=default
```

## Problem

Once I deployed minio example, all deployed ok. I can see deployed minio service in both `nomad` as a job and `consul` as a registered service.

But, healthchecks fail in ansible `dev/ansible/03_run_tests.yml`
## Log
```
TASK [Minio healthchecks pass] *************************************************
FAILED - RETRYING: Minio healthchecks pass (1 retries left).
fatal: [default]: FAILED! => {
    "attempts": 15,
    "changed": false,
    "connection": "close",
    "content": "[]\n",
    "content_length": "3",
    "content_type": "application/json",
    "cookies": {},
    "cookies_string": "",
    "date": "Tue, 29 Sep 2020 06:06:35 GMT",
    "elapsed": 0,
    "json": [],
    "redirected": false,
    "status": 200,
    "url": "http://localhost:8500/v1/health/checks/minio?filter=%28Status%3D%3D%22passing%22%29",
    "vary": "Accept-Encoding",
    "x_consul_effective_consistency": "leader",
    "x_consul_index": "120",
    "x_consul_knownleader": "true",
    "x_consul_lastcontact": "0"
}

MSG:

OK (3 bytes)

PLAY RECAP *********************************************************************
default                    : ok=12   changed=5    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0
```

## Why
- The reason, that `consul_acl_default_policy=deny` use white list concept. The administrator needs to explicitly manage who/what has access to what.
- [Healthchecks rest api](https://www.consul.io/api-docs/health#list-checks-for-service) does not fail on the **BAD** token (no token or token which is not authorized for the request). 
It basically returns an empty list of healthchecks.

## How to fix
Add token to HTTP request.

**Comment by zhenik** [comment]
The same applied to proxy. Once you need to create local proxy to service. Example:

```
consul connect proxy -service minio-local -upstream minio:9999 -log-level debug
```
Need `env` variable `CONSUL_HTTP_TOKEN`
```
CONSUL_HTTP_TOKEN=<token> consul connect proxy -service minio-local -upstream minio:9999 -log-level debug
```

**Comment by Neha-Sinha2305** [comment]
@zhenik : I have already worked upon it and i have the fix for it in the code. There are two other things that needed to be fixed in addition to above to get the tests passing:

1. Create intention allow from mc to minio before running the terraform.yml ansible file. As of now the mc job fails in nomad when run with **consul_acl=true** and **consul_default_policy=deny**, thus no buckets get created in minio. The terraform apply does not fail as there are no explicit healthcheck code in nomad mc hcl file. WE do not have explicit healthcheck because mc does not has it.
Logged error in mc nomad job:
```
mc: <ERROR> Unable to initialize new alias from the provided credentials. Get "http://127.0.0.1:9000/probe-bucket-sign-xykd6j3xnc2u/?location=": Connection closed by foreign host http://127.0.0.1:9000/probe-bucket-sign-xykd6j3xnc2u/?location=. Retry again.
```
2. Create intention allow from minio-local to minio before creating the proxy
**Note**: The explicit intention _allow_ needs to be created as by **default intention deny all** kicks is when **consul_default_policy is set to deny**.

**Comment by zhenik** [comment]
Working on. I will provide draft PR later today than we can discuss more precisely. Ok?
